docs/api.md

```diff
@@ -156,6 +156,8 @@ Note that there are three types of Hooks in TensorFlow: SessionHook, EstimatorHo
 | --- | --- | --- | --- |
 | `wrap_optimizer(optimizer)` | `optimizer` (tf.train.Optimizer or tf.keras.Optimizer) | Returns the same optimizer object passed with a couple of identifying markers to help `smdebug`. This returned optimizer should be used for training. | When not using Zero Script Change environments, calling this method on your optimizer is necessary for SageMaker Debugger to identify and save gradient tensors. Note that this method returns the same optimizer object passed and does not change your optimization logic. If the hook is of type `KerasHook`, you can pass in either an object of type `tf.train.Optimizer` or `tf.keras.Optimizer`. If the hook is of type `SessionHook` or `EstimatorHook`, the optimizer can only be of type `tf.train.Optimizer`.  This new
 | `add_to_collection(`<br/>  `collection_name, variable)` | `collection_name (str)` : name of the collection to add to. <br/> `variable` parameter to pass to the collection's `add` method. | `None` | Calls the `add` method of a collection object. See [this section](#collection) for more. |
+| `wrap_tape(tape)` | `tape` (tensorflow.python.eager.backprop.GradientTape) | Returns a tape object with three identifying markers to help `smdebug`. This returned tape should be used for training. | When not using Zero Script Change environments, calling this method on your tape is necessary for SageMaker Debugger to identify and save gradient tensors. Note that this method returns the same tape object passed.
```

Comment:
are these specific to TF2.x ?
I didn't get what is special in record_tensor_value that makes is only for TF2.x

Reply:
wrap_tape is specific to TF 2.x.

record_tensor_value() is generic, but restricted to metrics value.

Reply:
there is no regex for metrics. so by using this API, user can save any tensor but according to the check in this function, it will get saved as METRICS collection (if the user chooses to save metrics collection).

Reply:
did not create a generic save_tensor API as that will be a feature by itself.

used record_tensor_value() similar to the usage in PT hook for functional loss